blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/EnumMetric.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2013 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.types;
+
+import com.rackspacecloud.blueflood.exceptions.InvalidDataException;
+import com.rackspacecloud.blueflood.utils.TimeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
```

Comment:
@tilogaat great suggestion!  Please take a look at the new code here.  

I simplified your suggestions further and just added the "getStringEnumValues()" method in BluefloodEnumRollup, which returns the keySet of all enum values as an ArrayList<String>, and use the PreaggreatedMetric class instead of a new EnumMetric class to store and retrieve the metrics.

This means that we'll have to compose a batch of PreaggreatedMetric which include enum values to be called by EnumElasticIO.insertDiscovery(), instead of a batch of EnumMetrics.

I will see where to make that call when I am working on the DiscoveryWriter.